temp_functions.py

```diff
@@ -0,0 +1,75 @@
+import read_db
+import pandas as pd
+import time
+from tqdm import tqdm
+
+class functions():
+
+    def __init__(self):
+        self.df=None
+        self.dict={}
+
+    def set_dataframe(self):
+        extract = read_db.ExtractData()
+        self.df = extract.get_timeslots_info()
+
+    def main(self):
+        self.set_dataframe()
+        temp = {}
+        for i,row in tqdm(self.df.iterrows()):
+            key = row["EmployeeID"], row["RelationID"]
+
+            ## Calculate the remaining availability in each calendar month and week
+            appointmentLength = (row["UntilUtc"] - row["FromUtc"]).total_seconds()/3600
+            if key[0] in temp:
+                if temp[key[0]]["DateLastWorked"].year == row["FromUtc"].year:
+                    if temp[key[0]]["DateLastWorked"].month == row["FromUtc"].month:
+                        temp[key[0]]["HoursLeftInMonth"] = temp[key[0]]["HoursLeftInMonth"] - appointmentLength
+                    else: 
+                        temp[key[0]]["HoursLeftInMonth"] = row["AverageNumberOfHoursPerMonth"] - appointmentLength
+                    if temp[key[0]]["DateLastWorked"].isocalendar().week == row["FromUtc"].isocalendar().week:
+                        temp[key[0]]["HoursLeftInWeek"] = temp[key[0]]["HoursLeftInWeek"] - appointmentLength
+                    else: 
+                        temp[key[0]]["HoursLeftInWeek"] = row["NumberOfHoursPerWeek"] - appointmentLength
+                else: 
+                    temp[key[0]]["HoursLeftInMonth"] = row["AverageNumberOfHoursPerMonth"] - appointmentLength
+                    temp[key[0]]["HoursLeftInWeek"] = row["NumberOfHoursPerWeek"] - appointmentLength
+                temp[key[0]]["DateLastWorked"] = row["UntilUtc"]
+            else:
+                temp[key[0]] = {"HoursLeftInMonth": row["AverageNumberOfHoursPerMonth"] - appointmentLength, 
```

Comment:
Is dit inderdaad op basis van contract of is dit het average gewerkte uren?

Reply:
> Is dit inderdaad op basis van contract of is dit het average gewerkte uren?

Het is allebei op kalenderweek en kalendermaand afgewerkt, dit is een makkelijke fix en dit wilde ik de eerstvolgende keer dat we afspreken even opbrengen.

Reply:
Oke dus wel contract uren?

Reply:
Ik mis verder ook de link met meerdere contracten, het kon zijn dat iemand in 2019 8 uur per week werkte en 202 16 uur per week, dit is redelijk belanrgijk